app/models/user.rb

```diff
@@ -61,6 +62,25 @@ class User < ApplicationRecord
     :recoverable, :rememberable, :trackable, :validatable,
     :confirmable, :lockable, :timeoutable
 
+  # Defines a reusable mapping (CONSENT_ENUM) of consent constants
+  CONSENT_UNASKED = 'unasked'
+  CONSENT_CONSENTED = 'consented'
+  CONSENT_UNCONSENTED = 'unconsented'
+
+  CONSENT_ENUM = {
+    CONSENT_UNASKED => CONSENT_UNASKED,
+    CONSENT_CONSENTED => CONSENT_CONSENTED,
+    CONSENT_UNCONSENTED => CONSENT_UNCONSENTED
+  }.freeze
+
+  # enum :consent, {
+  #   CONSENT_UNASKED => CONSENT_UNASKED,
+  #   CONSENT_CONSENTED => CONSENT_CONSENTED,
+  #   CONSENT_UNCONSENTED => CONSENT_UNCONSENTED
+  # }, prefix: :consent
+
+  enum :contactable, CONSENT_ENUM, prefix: :contactable, validate: true
```

Comment:
would be useful to define the macros for solargraph like we did with the other enums

it would also demagic some of the macros